restapi/src/main/java/io/stargate/web/docsapi/service/write/DocumentWriteService.java

```diff
@@ -237,6 +237,71 @@ public DocumentWriteService(TimeSource timeSource, DocsApiConfiguration config)
         .observeOn(Schedulers.io());
   }
 
+  /**
+   * Deletes a single whole document.
+   *
+   * @param dataStore {@link DataStore}
+   * @param keyspace Keyspace to delete a document from.
+   * @param collection Collection the document belongs to.
+   * @param documentId Document ID.
+   * @param rows Rows of this document.
+   * @param numericBooleans If numeric boolean should be stored.
+   * @param context Execution content for profiling.
+   * @return Single containing the {@link ResultSet} of the batch execution.
+   */
+  public Single<ResultSet> deleteDocument(
+      DataStore dataStore,
+      String keyspace,
+      String collection,
+      String documentId,
+      ExecutionContext context) {
+    List<String> subDocumentPath = Collections.emptyList();
+    return deleteDocument(dataStore, keyspace, collection, documentId, subDocumentPath, context);
+  }
+
+  /**
+   * Deletes a single (sub-)document, ensuring that existing document with the same key have the
+   * given sub-path deleted.
+   *
+   * @param dataStore {@link DataStore}
+   * @param keyspace Keyspace to delete a document from.
+   * @param collection Collection the document belongs to.
+   * @param documentId Document ID.
+   * @param subDocumentPath The sub-document path to delete.
+   * @param context Execution content for profiling.
+   * @return Single containing the {@link ResultSet} of the batch execution.
+   */
+  public Single<ResultSet> deleteDocument(
+      DataStore dataStore,
+      String keyspace,
+      String collection,
+      String documentId,
+      List<String> subDocumentPath,
+      ExecutionContext context) {
+    // create and cache the remove query prepare
+    Single<? extends Query<? extends BoundQuery>> deleteQueryPrepare =
+        prepareDeleteDocumentQuery(dataStore, keyspace, collection, subDocumentPath);
+
+    // execute when prepared
+    return deleteQueryPrepare
+        .observeOn(Schedulers.computation())
```

Comment:
I'm curious why we want this operation to run on a "compuation" thread pool? Why not run it all on the "io" thread pool? Wouldn't this result in migration from one type to the other type of thread?

Reply:
It's to do binding op off the TPC thread (DSE) or Native Transport Thread (C*).. Yes it will offload to another thread.. We confirmed in the perf tests and JMH tests that there is not much of a difference, if any.. However, if we don't go away from those "storage" threads then they will be doing any task that is in the followup of this reactive sequence.. Meaning even logging the output of the request or writing to the request buffer..

Reply:
So this is a way to avoid blocking?

Reply:
It;s to avoid any extra work being done on the TCP thread and a safety net of not invoking new data store calls on this thread, which has caused issues in the past..